crates/wasm/src/view_server.rs

```diff
@@ -625,3 +533,42 @@ pub fn load_tree(stored_tree: StoredTree) -> Tree {
     let tree = add_hashes.finish();
     return tree;
 }
+
+pub async fn store_advice(note: Note) {
+    let db_req: OpenDbRequest = IdbDatabase::open_u32("penumbra", 12).ok().unwrap();
+
+    let db: IdbDatabase = db_req.into_future().await.ok().unwrap();
+
+    let tx = db.transaction_on_one("notes").ok().unwrap();
+    let store = tx.object_store("notes").ok().unwrap();
```

Comment:
I noticed we use a lot of unwrap()'s. Also these patterns:

```rust
.ok()?.await.ok()?;
.ok().unwrap()
```

In general, we need to move to the standard `Result` pattern and not use unwraps unless they are right as we coerce them to a `JsValue` and encode them for javascript.

Also, doing `.ok()` on everything has downsides:
- We lose error information, making debugging harder
- `?` is usually used on Result types. Using it on an Option is confusing.
- This will cause a panic if a None is received.
- We are treating Options as Results.

Reply:
In most cases we will have to use a rather massive construction with `map_err`
` .map_err(|err| JsError::new(&err.to_string()))?`
On this [issue](https://github.com/rustwasm/wasm-bindgen/issues/1742) , there is no more ergonomic solution yet